#!/usr/bin/env python
PACKAGE = "jsk_mbzirc_tasks"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#max
gen.add("hsv_h_max",    int_t,    0, "HSi Hue max param",  120,  0, 180)
gen.add("hsv_s_max",    int_t,    0, "HSi Saturation max param",  200,  0, 255)
gen.add("hsv_v_max",    int_t,    0, "HSi Brightless max param",  255,  0, 255)
#min 
gen.add("hsv_h_min",    int_t,    0, "HSi Hue min param",  0,  0, 180)
gen.add("hsv_s_min",    int_t,    0, "HSi Saturation min param",  101,  0, 255)
gen.add("hsv_v_min",    int_t,    0, "HSi Brightless min param",  80,  0, 255)
#clustering parameters
gen.add("cluster_threhold_dist",    int_t,    0, "cluster pixel distance",  18,  0, 400)
gen.add("downsample_cluster_size",    int_t,    0, "threshold for the clusteres to be downsampled",  1000,  100, 5000)
gen.add("min_cluster_size",    int_t,    0, "small clusteres to be remove",  10,  0, 1000)

exit(gen.generate(PACKAGE, "jsk_mbzirc_tasks", "Filterparam"))
